### ISSUE TITLE
Make tags as sha1 only before save or optional?

### ISSUE THREAD

**Issue by Kudja** [GordonLesti/Lesti_Fpc#216]
Every method that returns tags immidiatly make them as sha1, but it is not very useful if I want to update them via fpc_observer_collect_cache_tags event for example.

So in some cases I want to remove all cms blocks tags from tags list - so I have no need to refresh page on this blocks changed (it would be refreshed in other way), but as all tags in sha1 I can't filter them

May be remove all that sha1... and make arra_map for sha1 just before save?

**Comment by GordonLesti** [comment]
Yes, this was really a stupid decision in the past. I'm sorry. I would like to change that, but many people at the moment rely on `save` & `clean` are taking `sha1` of a tag. Maybe it's time for a breaking version?

**Comment by Kudja** [comment]
May be such solution
1. replace all calls of sha1 to some custom function like fpc_sha
2. Add some option to admin config like tags_save_mode (default value to make them sha1 directly on collect functions)
3. Mark methods that collects options as deprecated
4. in custom function check config flag and based on it make tags sha1 directly, or just before saving them...

This is not very elegant, but this can work + on next one-two updates you can remove that deprecated code and option
